### PR TITLE
chore: Update Wave component to use WaveComponent type

### PR DESCRIPTION
**Pull request by baicie**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...
optimization wave type
- [ ] 🆕 New feature
- [ ] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [x] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Optimize wave supported component types        |
| 🇨🇳 Chinese |  优化wave支持组件类型         |

### CODE REVIEW

**Review by Wxh16144 — COMMENTED** [review]
ShowWaveEffect.info.component 是不是也可以顺手加上🤔

**Comment by baicie** [comment]
> ShowWaveEffect.info.component 是不是也可以顺手加上🤔

我看看加一下

File: components/_util/wave/interface.ts [hunk 1]
```diff
@@ -8,10 +8,12 @@ export type ShowWaveEffect = (
   info: {
     className: string;
     token: GlobalToken;
-    component?: string;
+    component?: WaveComponent;
```
Comment (li-jia-nan) [inline components/_util/wave/interface.ts:11]: 这个类型 export 出去了，从宽的范围收到窄的范围，会不会对历史用户造成 breaking change ？

**Comment by afc163** [comment]
改动这个对用户的影响是？

File: components/_util/wave/interface.ts (hunk 1 above)
Comment (baicie) [reply components/_util/wave/interface.ts:11]: 我看了下代码，因为antd支持wave效果的只有这些组件，所以component本该只有这些取值，如果出现错误正好是发现了不合法的使用。不知道我的理解有没有问题

**Comment by baicie** [comment]
> 改动这个对用户的影响是？

我看了下代码，因为antd支持wave效果的只有这些组件，所以component本该只有这些取值，如果出现错误正好是发现了不合法的使用。不知道我的理解有没有问题

**Review by afc163 — APPROVED** [review]

**Pull request merged**